static/app/views/alerts/list/rules/alertRulesList.tsx

```diff
@@ -278,9 +282,8 @@ function AlertRulesList() {
                         key={`${
                           isIssueAlert(rule) ? AlertRuleType.METRIC : AlertRuleType.ISSUE
                         }-${rule.id}`}
-                        showMigrationWarning={
-                          hasMigrationUIFeatureFlag && ruleNeedsMigration(rule)
-                        }
+                        showMigrationWarning={rulesWithMigrationWarnings.has(rule)}
```

Comment:
nit: Any advantage to using Sets instead of `ruleIdsWithmigrationWarnings.includes(rule.id)`?

Reply:
lookup runtime complexity